src/neuroconv/utils/dict.py

```diff
@@ -198,3 +198,24 @@ def dict_deep_update(
             dict_to_update[key_to_update] = update_values
 
     return dict_to_update
+
+
+class DeepDict(dict):
+    """A defaultdict of defaultdicts"""
+
+    def __getitem__(self, item):
+        try:
+            return dict.__getitem__(self, item)
+        except KeyError:
+            self[item] = DeepDict()
+            return self[item]
```

Comment:
This looks very useful generally; only thing, could you add some unit tests that showcase the base expected usage of this class? Such as

1) test that demonstrates commands like

```python
my_deep_dict["key1"]["key2"]["key3"] = some_value
```

2) can keys at any level be any hashable type? I.e.,

```python
my_deep_dict["key1"][2][(3,)] = some_value
```

3) Demonstration of the `print` (indirect call to `repr`) of an instance of such a deep dict (one that is say, 3 levels deep with around ~6 items injected at various levels

4) optional, but encourage at least one test showcasing raising of the `KeyError` at a deep level

Reply:
1: done
2: done
3: done
4: the function is refactored so this is no longer relevant